lib/page/prompt.js

```diff
@@ -57,76 +69,119 @@ function promptRegister() {
 
 	//create input/select/counter/keybind
 	const dataContainerElement = $("#data-container");
+	const buttonsContainer = $("#buttons");
 
-	switch (promptOptions.type) {
+	switch (promptOptions.type)
+	{
 		case "counter":
 			dataElement = promptCreateCounter(dataContainerElement);
 			break;
 		case "input":
-			dataElement = promptCreateInput();
+			dataElement = promptCreateInput(promptOptions);
 			break;
 		case "select":
 			dataElement = promptCreateSelect();
 			break;
 		case "keybind":
 			dataElement = require("./keybind")(promptOptions.keybindOptions, dataContainerElement);
 			break;
+		case "multiInput":
+			dataElement = promptCreateMultiInput(dataContainerElement);
+			break;
 		default:
 			return promptError(`Unhandled input type '${promptOptions.type}'`);
 	}
 
-	if (promptOptions.type != "keybind") {
+	if (promptOptions.type != "keybind" && promptOptions.type != "multiInput")
+	{
 		dataElement.setAttribute("id", "data");
 
-		if (promptOptions.type !== "counter") {
+		if (promptOptions.type !== "counter")
+		{
 			dataContainerElement.append(dataElement);
 		}
 	}
 
 	dataElement.focus();
 
-	if (promptOptions.type !== "select" && promptOptions.type !== "keybind") {
+	if (promptOptions.type !== "select" && promptOptions.type !== "keybind")
+	{
 		dataElement.select();
 	}
 
 	//load custom script from options
-	if (promptOptions.customScript) {
-		try {
+	if (promptOptions.customScript)
+	{
+		try
+		{
 			const customScript = require(promptOptions.customScript);
 			customScript();
-		} catch (error) {
+		} catch (error)
+		{
 			return promptError(error);
 		}
 	}
+
+	if (promptOptions.button)
+	{
+		const button = document.createElement("button");
+
+		// insert custom input attributes if in options
+		if (promptOptions.button.attrs && typeof (promptOptions.button.attrs) === "object")
+		{
+			for (const k in promptOptions.button.attrs)
+			{
+				if (!Object.prototype.hasOwnProperty.call(promptOptions.button.attrs, k))
+				{
+					continue;
+				}
+
+				button.setAttribute(k, promptOptions.button.attrs[k]);
+			}
+		}
+		button.addEventListener("click", () => eval(promptOptions.button.click));
+		button.innerText = promptOptions.button.label;
+		button.setAttribute("type", "button");
+		button.setAttribute("id", "custom");
+
+		buttonsContainer.append(button);
```

Comment:
I think it might be better to move the custom button to be either
* between the cancel button and the ok one.
* to the left of the cancel button

  could be easily done with:
  ```js
  $("#cancel").after(button);
  //or
  buttonsContainer.prepend(button);
  //or
  $("#cancel").before(button);

  ```